test/concurrent/SynchronizedListTest.ooc

```diff
@@ -0,0 +1,328 @@
+			result := list getSlice(1 .. 3)
+			expect(result count == 3)
+			expect(result[0] == list[1])
+			expect(result[1] == list[2])
+			expect(result[2] == list[3])
+			result clear()
+			result = list getSlice(1, 3)
+			expect(result count == 3)
+			expect(result[0] == list[1])
+			expect(result[1] == list[2])
+			expect(result[2] == list[3])
+			list free()
+			result free()
+		})
+		this add("single thread - getSliceInto", func {
+			list := SynchronizedList<Float> new()
+			list add(1.0f) . add(2.0f) . add(3.0f) . add(4.0f)
+			sliceInto := VectorList<Float> new()
+			list getSliceInto(Range new(1, 2), sliceInto)
+			expect(sliceInto[0], is equal to(2.0f) within(0.00001f))
+			expect(sliceInto[1], is equal to(3.0f) within(0.00001f))
+			list free()
+			sliceInto free()
+		})
+		this add("single thread - operators", func {
+			list := SynchronizedList<Int> new()
+			list add(-5) . add(-4) . add(-3) . add(-2) . add(-1)
+			oldValue := list[0]
+			list[0] = -6
+			expect(oldValue == -5)
+			expect(list[0] == -6)
+			list free()
+		})
+		this add("single thread (class)", func {
+			count := 10_000
+			list := SynchronizedList<Cell<ULong>> new()
+			for (value in 0 .. count) {
+				list add(Cell<ULong> new(value))
+				expect(list count, is equal to(value + 1))
+				expect(list[value] get(), is equal to(value))
+			}
+			for (value in 0 .. count) {
+				resultValue := list remove()
+				expect(resultValue get() , is equal to (count - value - 1))
+				resultValue free()
+			}
+			list free()
+		})
+		this add("single thread - iterator", func {
+			list := SynchronizedList<Int> new()
+			list add(8) . add(16) . add(32)
+			iterator := list iterator()
+			expect(iterator hasNext?(), is true)
+			for ((index, item) in iterator)
+				expect(item, is equal to(list[index]))
+			expect(iterator hasNext?(), is false)
+			secondIterator := list iterator()
+			expect(secondIterator next(), is equal to(8))
+			secondIterator free()
+			iterator free()
+			list free()
+		})
+		this add("multiple threads", This _testMultipleThreads)
+		this add("multiple threads (class)", This _testMultipleThreadsWithClass)
+	}
+	_testMultipleThreads: static func {
+		numberOfThreads := 8
+		countPerThread := 10_000
+		list := SynchronizedList<Int> new()
+		threads := Thread[numberOfThreads] new()
+		job := func {
+			for (i in 0 .. countPerThread)
+				list add(i)
+		}
+		for (i in 0 .. numberOfThreads) {
+			threads[i] = Thread new(job)
+			threads[i] start()
+		}
+		for (i in 0 .. numberOfThreads) {
+			threads[i] wait()
+			threads[i] free()
+		}
+		expect(list count, is equal to(numberOfThreads * countPerThread))
+		(job as Closure) free()
+		job = func {
+			for (i in 0 .. countPerThread) {
+				value := list remove(0)
+				expect(value >= 0 && value < countPerThread)
+			}
+		}
+		for (i in 0 .. numberOfThreads) {
+			threads[i] = Thread new(job)
+			threads[i] start()
+		}
+		for (i in 0 .. numberOfThreads) {
+			threads[i] wait()
+			threads[i] free()
```

Comment:
You can do: `threads[i] wait() . free()` here and in other places in this test.
Also, 
```ooc
threads[i] = Thread new(job)
threads[i] start()
```
can probably be written `threads[i] = Thread new(job) . start()`